src/Util/ExAuth.php

```diff
@@ -0,0 +1,305 @@
+ * 	  registering with any client so they should be instructed to replace these chars
+ * 	  " " (space) is replaced with "%20"
+ * 	  "@" is replaced with "(a)"
+ *
+ */
+
+namespace Friendica\Util;
+
+use Friendica\Core\Config;
+use Friendica\Core\PConfig;
+use Friendica\Database\DBM;
+use dba;
+
+require_once 'include/dba.php';
+
+class ExAuth
+{
+	private $bDebug;
+
+	/**
+	 * @brief Create the class
+	 *
+	 * @param boolean $bDebug Debug mode
+	 */
+	public function __construct()
+	{
+		$this->bDebug = (int) Config::get('jabber', 'debug');
+
+		openlog('auth_ejabberd', LOG_PID, LOG_USER);
+
+		$this->writeLog(LOG_NOTICE, 'start');
+	}
+
+	/**
+	 * @brief Standard input reading function, executes the auth with the provided
+	 * parameters
+	 *
+	 * @return null
+	 */
+	public function readStdin()
+	{
+		while (!feof(STDIN)) {
+			// Quit if the database connection went down
+			if (!dba::connected()) {
+				$this->writeLog(LOG_ERR, 'the database connection went down');
+				return;
+			}
+
+			$iHeader = fgets(STDIN, 3);
+			$aLength = unpack('n', $iHeader);
+			$iLength = $aLength['1'];
+
+			// No data? Then quit
+			if ($iLength == 0) {
+				$this->writeLog(LOG_ERR, 'we got no data, quitting');
+				return;
+			}
+
+			// Fetching the data
+			$sData = fgets(STDIN, $iLength + 1);
+			$this->writeLog(LOG_DEBUG, 'received data: ' . $sData);
+			$aCommand = explode(':', $sData);
+			if (is_array($aCommand)) {
+				switch ($aCommand[0]) {
+					case 'isuser':
+						// Check the existance of a given username
+						$this->isUser($aCommand);
+						break;
+					case 'auth':
+						// Check if the givven password is correct
+						$this->auth($aCommand);
+						break;
+					case 'setpass':
+						// We don't accept the setting of passwords here
+						$this->writeLog(LOG_NOTICE, 'setpass command disabled');
+						fwrite(STDOUT, pack('nn', 2, 0));
+						break;
+					default:
+						// We don't know the given command
+						$this->writeLog(LOG_NOTICE, 'unknown command ' . $aCommand[0]);
+						fwrite(STDOUT, pack('nn', 2, 0));
+						break;
+				}
+			} else {
+				$this->writeLog(LOG_NOTICE, 'invalid command string ' . $sData);
+				fwrite(STDOUT, pack('nn', 2, 0));
+			}
+		}
+	}
+
+	/**
+	 * @brief Check if the given username exists
+	 *
+	 * @param array $aCommand The command array
+	 */
+	private function isUser(array $aCommand)
+	{
+		$a = get_app();
+
+		// Check if there is a username
+		if (!isset($aCommand[1])) {
+			$this->writeLog(LOG_NOTICE, 'invalid isuser command, no username given');
+			fwrite(STDOUT, pack('nn', 2, 0));
+			return;
+		}
+
+		// Now we check if the given user is valid
+		$sUser = str_replace(array('%20', '(a)'), array(' ', '@'), $aCommand[1]);
+
+		// Does the hostname match? So we try directly
+		if ($a->get_hostname() == $aCommand[2]) {
+			$this->writeLog(LOG_INFO, 'internal user check for ' . $sUser . '@' . $aCommand[2]);
+			$found = dba::exists('user', ['nickname' => $sUser]);

```

Comment:
Maybe a dumb question, but: is ```array('nickname' => $sUser)``` the same like ```['nickname' => $sUser]```?

Reply:
It is since PHP 5 and change!